src/gwt/src/org/rstudio/studio/client/workbench/views/source/editors/text/rmd/display/DefaultChunkOptionsPopupPanel.java

```diff
@@ -224,14 +308,152 @@ else if (modeId == "mode/r")
             endIndex = cursor.getIndex();
          }
 
+         String optionName = StringUtil.substring(arguments, startIndex, equalsIndex).trim();
          chunkOptions.put(
-               StringUtil.substring(arguments, startIndex, equalsIndex).trim(),
-               StringUtil.substring(arguments, equalsIndex + 1, endIndex).trim());
+               ChunkOptionValue.normalizeOptionName(optionName),
+               new ChunkOptionValue(
+                  StringUtil.substring(arguments, equalsIndex + 1, endIndex).trim(),
+                  OptionLocation.FirstLine));
 
          startIndex = cursor.getIndex() + 1;
       }
       while (cursor.moveToNextCharacter());
    }
 
+   /**
+    * Returns list of option lines from the editor; these follow the first line,
+    * must begin with "#| ", and must have at least one non-whitespace character
+    * after "#| ". The lines aren't checked to see if they contain valid YAML, and
+    * the option names are not normalized (e.g. fig.cap vs. fig-cap).
+    * 
+    * @param startLine first potential option line
+    * @return list of option lines without the leading #|
+    */
+   private List<String> getYamlOptionLines(int startLine)
+   {
+      List<String> optionLines = new ArrayList<>();
+      int currentLine = startLine;
+      String line;
+      while ((line = display_.getLine(currentLine)) != null)
+      {
+         // a line with no characters after the starting #| isn't valid and ends detection
+         if (line.startsWith("#| ") && line.trim().length() > 2)
+         {
+            optionLines.add(line.substring("#| ".length()));
+            currentLine++;
+         }
+         else
+         {
+            break;
+         }
+      }
+      return optionLines;
+   }
+
+   /**
+    * Returns count of option lines in the editor; these follow the first line and
+    * must begin with "#|". No validation of line contents after the prefix is performed.
+    * 
+    * @param startLine first potential option line
+    * @return number of lines found
+    */
+   private int countYamlOptionLines(int startLine)
+   {
+      int currentLine = startLine;
+      String line;
+      while ((line = display_.getLine(currentLine)) != null)
+      {
+         if (line.startsWith("#|"))
```

Comment:
It looks like the behavior here might differ between Quarto and R Markdown. In particular, R Markdown requires some whitespace following `#|`; e.g.

`````r
```{r}
#|example: true
print(knitr::opts_current$get("example"))
```
`````

When rendered, this will print "NULL" rather than TRUE. I'm not sure if this is true for Quarto as well, though.

Again, not sure if this is actually worth worrying about in practice, though!